discovery-infra/test_infra/helper_classes/kube_helpers/agent_cluster_install.py

```diff
@@ -146,9 +146,32 @@ def _get_spec_dict(
         if "machine_cidr" in kwargs:
             spec["networking"]["machineNetwork"] = [{"cidr": kwargs.pop("machine_cidr")}]
 
+        if "hyperthreading" in kwargs:
+            self._set_hyperthreading(spec=spec, mode=kwargs.pop("hyperthreading"))
+
         spec.update(kwargs)
         return spec
 
+    def _set_hyperthreading(self, spec: Dict[str, Any], mode: str) -> None:
+        # if hypethreading is not configured, let the service choose the default setup
+        if not mode:
```

Comment:
i think you can remove it, since you check that it is present in line 149

Reply:
No, In line 149 I check that the --hyperthreading option is there but not its value. It is true that now the default value of this option will not be None if we are reaching this point from day1_flow() but since it is going to be removed I left this check to cover all options.